### PR TITLE
Update xunit version to match corefx

### PR DESCRIPTION
**Pull request by jkotas**
Fixes https://github.com/dotnet/corefx/issues/33587

### CODE REVIEW

File: tests/src/Common/CoreFX/CoreFX.depproj [hunk 1]
```diff
@@ -36,7 +37,7 @@
       <Version>$(CoreFXXUnitPackageVersion)</Version>
     </PackageReference>
     <PackageReference Include="xunit.assert">
```
Comment (jkotas) [inline tests/src/Common/CoreFX/CoreFX.depproj]: Updating everything looks pretty non-trivial. So I am only updating the part that we need.

File: tests/src/Common/CoreFX/CoreFX.depproj [hunk 2]
```diff
@@ -13,7 +13,8 @@
       <SystemCompositionVersions>1.3.0-preview3-26501-04</SystemCompositionVersions>
       <XUnitNetcoreExtensionsVersion>2.2.0-preview1-02902-01</XUnitNetcoreExtensionsVersion>
       <!-- Xunit Abstractions is differently versioned then the rest of XUnit -->
-      <XUnitAbstractionsVersion>2.0.1</XUnitAbstractionsVersion>
+      <XUnitAbstractionsVersion>2.0.3</XUnitAbstractionsVersion>
+      <XUnitAssertVersion>2.4.1-pre.build.4059</XUnitAssertVersion>
```
Comment (stephentoub) [inline tests/src/Common/CoreFX/CoreFX.depproj]: Thanks, Jan.  I assume there was a change in Assert.Equal where previously (in the version coreclr has) it was assuming GetType() == GetType(), and now it's not requiring that?

Comment (jkotas) [reply tests/src/Common/CoreFX/CoreFX.depproj]: Right, here is the related change from xunit: https://github.com/xunit/assert.xunit/commit/a42b055ae6fbaf2b54df079bbf3c51404a99d2c7#diff-445c3deeda8978a1cfcb9ec2b9f45d8eL52

**Comment by jkotas** [comment]
Array.Copy test is failing because of this: https://github.com/dotnet/corefx/pull/31260/commits/591f0b5ba2c0ddcdad1a326eb327936cb2dd40d8#diff-5b0c2a1222da27a6933bd14cfdfe7f01L1000

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT x64 Checked Innerloop Build and Test (Jit - TieredCompilation=0) please

**Pull request merged**